src/Tools/PrepareTests/TestDiscovery.cs

```diff
@@ -85,14 +86,22 @@ static bool RunWorker(string dotnetPath, string pathToWorker, string pathToAssem
             pipeClient.StartInfo.FileName = pathToWorker;
         }
 
+        var errorOutput = new StringBuilder();
+
         using (var pipeServer = new AnonymousPipeServerStream(PipeDirection.Out, HandleInheritability.Inheritable))
         {
             // Pass the client process a handle to the server.
             arguments.Add(pipeServer.GetClientHandleAsString());
             pipeClient.StartInfo.Arguments = string.Join(" ", arguments);
             pipeClient.StartInfo.UseShellExecute = false;
+
+            // Errors will be logged to stderr, redirect to us so we can capture it.
```

Comment:
errors were getting swallowed so it was just failing with no info.  this ensures error output from the discovery worker is captured